app/src/main/java/de/rochefort/childmonitor/ListenService.kt

```diff
@@ -0,0 +1,212 @@
+            val name = extras.getString("name")
+            childDeviceName = name
+            val n = buildNotification(name)
+            val foregroundServiceType = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK else 0 // Keep the linter happy
+            ServiceCompat.startForeground(this, ID, n, foregroundServiceType)
+            val address = extras.getString("address")
+            val port = extras.getInt("port")
+            doListen(address, port)
+        }
+        return START_REDELIVER_INTENT
+    }
+
+    override fun onDestroy() {
+        val lt = this.listenThread
+        if (lt != null) {
+            lt.interrupt()
+            this.listenThread = null
+        }
+
+        // Cancel the persistent notification.
+        val NOTIFICATION = R.string.listening
+        notificationManager!!.cancel(NOTIFICATION)
+        ServiceCompat.stopForeground(this, ServiceCompat.STOP_FOREGROUND_REMOVE)
+        // Tell the user we stopped.
+        Toast.makeText(this, R.string.stopped, Toast.LENGTH_SHORT).show()
+    }
+
+    override fun onBind(intent: Intent): IBinder {
+        return binder
+    }
+
+    private fun buildNotification(name: String?): Notification {
+        // In this sample, we'll use the same text for the ticker and the expanded notification
+        val text = getText(R.string.listening)
+
+        // The PendingIntent to launch our activity if the user selects this notification
+        val contentIntent = PendingIntent.getActivity(this, 0,
+                Intent(this, ListenActivity::class.java), PendingIntent.FLAG_IMMUTABLE)
+
+        // Set the info for the views that show in the notification panel.
+        val b = NotificationCompat.Builder(this, CHANNEL_ID)
+        b.setSmallIcon(R.drawable.listening_notification) // the status icon
+                .setOngoing(true)
+                .setTicker(text) // the status text
+                .setContentTitle(text) // the label of the entry
+                .setContentText(name) // the contents of the entry
+                .setContentIntent(contentIntent)
+        return b.build()
+    }
+
+    private fun createNotificationChannel() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val serviceChannel = NotificationChannel(
+                    CHANNEL_ID,
+                    "Foreground Service Channel",
+                    NotificationManager.IMPORTANCE_DEFAULT
+            )
+            notificationManager!!.createNotificationChannel(serviceChannel)
+        }
+    }
+
+    fun setErrorCallback(errorCallback: Runnable?) {
+        mErrorCallback = errorCallback
+    }
+
+    fun setUpdateCallback(updateCallback: Runnable?) {
+        mUpdateCallback = updateCallback
+    }
+
+    inner class ListenBinder : Binder() {
+        val service: ListenService
+            get() = this@ListenService
+    }
+
+    private var mErrorCallback: Runnable? = null
```

Comment:
Unrelated to kotlin, but can we get rid of those m prefixes? 

also here don't use Runnables

Reply:
Done both.